### ISSUE TITLE
Disable certain Taps and gestures action for File browser

### ISSUE THREAD

**Issue by shuvashish76** [koreader/koreader#12111]
I don't see the use of **All** Taps and gesture actions related to 1. Reader, 2. Reflowable documents, 3. Fixed layout documents for "File browser". I tried most of them which does nothing, I guess better we should disable them?
Note: Do let me know if any of the specific sub-action works on File browser hence should be enabled.

**Comment by mergen3107** [comment]
I don't know what defaults are for FM, so I just removed ALL gestures and taps and then started fresh including only those options that make sense for FM :D

**Comment by pazos** [comment]
Everything works as expected.

Gesture categories are designed as preconditions, so everything but "general" and "device" has a precondition (reader?, FM?, reflowable? fixed?...)

They're already disabled if the precondition isn't met.

If you mean remove them from UI, that's a nightmare :p.

We should provide the same set of actions for a given device. This makes possible to setup the same gesture with various actions based on the preconditions. With your suggestion I would need to switch between reader and FM to setup what the gesture does on that specific mode.

**Comment by mergen3107** [comment]
As a quickest corner cut, we can just add “Reader only: “ in the beginning of “reflowable” and “fixed layout” categories

**Comment by shuvashish76** [comment]
> If you mean remove them from UI, that's a nightmare :p.

Yes suggestion is similar to how we've disabled 1. Tap corner (Tap left/right), 2. All Double Tap actions, 3. Long Tap on corner (Top left) (fixed in nightly build) for **File Browser** only.

> Gesture categories are designed as preconditions, so everything but "general" and "device" has a precondition (reader?, FM?, reflowable? fixed?...)

Sorry I don't get your point...what do you mean by "preconditions"? 

>  With your suggestion I would need to switch between reader and FM to setup what the gesture does on that specific mode.

Isn't it already don't that way?, I've different actions for the same gestures specific to FM & Reader 🤔

**Comment by pazos** [comment]
> Sorry I don't get your point...what do you mean by "preconditions"?

You need to be on the FM to fire FM actions. On the reader to fire reader actions. On a reflowable document to fire reflowable document actions. On fixed page layout document to fire fixed page layout document actions.

> Isn't it already done that way?, I've different actions for the same gestures specific to FM & Reader 🤔

Yes, and you can setup a gesture for both the FM & Reader at the same time. That won't be possible with your suggestion.

**Comment by pazos** [comment]
Maybe I'm missing something, I hardly now the subject :smiling_face_with_tear:

**Comment by mergen3107** [comment]
FM and Reader gesture sets are independent of each other, by the way. 

So you need to set up both sets in full separately.

**Comment by shuvashish76** [comment]
> Yes, and you can setup a gesture for both the FM & Reader at the same time.

How can I do that? I never noticed it...I thought both are independent of each other as always.
For clarity I'm saying that you can't setup gestures for Reader while you're in FM and vice versa.
